packages/salesforcedx-vscode-core/src/conflict/persistentStorageService.ts

```diff
@@ -35,22 +35,26 @@ export class PersistentStorageService {
     return PersistentStorageService.instance;
   }
 
-  public getPropertiesForFile(fileName: string): ConflictFileProperties | undefined {
-    return this.storage.get<ConflictFileProperties>(fileName);
+  public getPropertiesForFile(key: string): ConflictFileProperties | undefined {
+    return this.storage.get<ConflictFileProperties>(key);
   }
 
-  public setPropertiesForFile(fileName: string, value: ConflictFileProperties | undefined) {
-    this.storage.update(fileName, value);
+  public setPropertiesForFile(key: string, conflictFileProperties: ConflictFileProperties | undefined) {
+    this.storage.update(key, conflictFileProperties);
   }
 
-  public setPropertiesForFiles(fileProperties: FileProperties | FileProperties[]) {
+  public setPropertiesForFilesRetrieve(fileProperties: FileProperties | FileProperties[]) {
     const fileArray = Array.isArray(fileProperties) ? fileProperties : [fileProperties];
     for (const fileProperty of fileArray) {
       this.setPropertiesForFile(
-        fileProperty.fileName,
+        this.makeKey(fileProperty.type, fileProperty.fullName),
```

Comment:
@violetyao @maxcamp So a component can be mapped to multiple files, which is where I'm having trouble understanding using this as a key for a single file. I might be missing some context though if you wouldn't mind breaking it down for me :)

Reply:
Great question! Max and I think the cache should save for each component because we don't have info on `lastModifiedDate` about individual files both on deploy and retrieve. Specifically, on the retrieve side (`setPropertiesForFilesRetrieve` ), we extract `lastModifiedDate` from `fileProperties` and `fileProperties` is for each single component; on the deploy side (`setPropertiesForFilesDeploy` ), we extract `lastModifiedDate` from `MetadataApiDeployStatus` and assign it to all components. Let us know what you think! :slightly_smiling_face:

Reply:
I see, that approach definitely makes sense. I just reviewed the API documentation [here](https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_result_objects.htm) to add some more context to your response. Thank you!